src/PureScript/Backend/Optimizer/Semantics.purs

```diff
@@ -613,12 +613,10 @@ evalAssocLet :: Env -> BackendSemantics -> (Env -> BackendSemantics -> BackendSe
 evalAssocLet env sem go = case sem of
   SemLet ident val k ->
     SemLet ident val \nextVal1 ->
-      makeLet Nothing (k nextVal1) \nextVal2 ->
-        go (bindLocal (bindLocal env (One nextVal1)) (One nextVal2)) nextVal2
+      evalAssocLet env (k nextVal1) go
```

Comment:
I'm not sure this is recursive in the right way. I think to reassociate left-associated lets, would you want to recurse greedily on `val` rather than under the continuation?